[Querier] Deprecate `-querier.compress-http-responses-deprecated` (#3544)
cortexproject/cortex: CHANGELOG.md

## master / unreleased

> * [CHANGE] Query Frontend: depcrated `-querier.compress-http-responses-deprecated`. You should use `-api.response-compression-enabled` instead. #3544

Reviewer: ```suggestion
* [CHANGE] Query Frontend: deprecate `-querier.compress-http-responses-deprecated` in favor of `-api.response-compression-enabled`. #3544
```

Reviewer 2: The flag appears to be named `-querier.compress-http-responses` ?
